test/index.js

```diff
@@ -67,33 +64,34 @@ describe('mkdirpStream', function () {
     });
   });
 
-  it('exports a main function, .obj and .withMode methods', function (done) {
+  it('exports a main function', function (done) {
     expect(typeof mkdirpStream).toEqual('function');
-    expect(typeof mkdirpStream.obj).toEqual('function');
     done();
   });
 
   it('takes a string to create', function (done) {
-    function assert() {
+    function assert(err) {
       expect(statMode(outputDirpath)).toBeDefined();
+      done(err);
     }
 
-    pipe([from(['test']), mkdirpStream(outputDirpath), concat(assert)], done);
+    pipeline(Readable.from(['test']), mkdirpStream(outputDirpath), assert);
```

Comment:
This only works because we didn't hit the `hwm`. All the tests should end on a writable stream to be "correct"

Reply:
Confirmed this is correct and piping to `new Writable()` fixes the issue.  Will push update.